modules/extension/xsd/xsd-gml3/src/main/java/org/geotools/gml3/simple/PointEncoder.java

```diff
@@ -54,7 +54,7 @@ public void encode(Point geometry, AttributesImpl atts, GMLWriter handler, Strin
         handler.startElement(pos, null);
 
         Coordinate coordinate = geometry.getCoordinate();
-        handler.position(coordinate.x, coordinate.y, coordinate.z);
+        handler.position(coordinate.x, coordinate.y, coordinate.z, coordinate.getM());
```

Comment:
Please use coordiante.getZ(), direct field access has been marked deprecated (especially since coordinate may not have a Z value.

I cannot remember if coordinate.getZ() will return NaN if geometry dimension is 2, or an exception.

Checking [CoordinateXY](https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/geom/CoordinateXY.java#L62) ... getZ() returns NULL_ORDINATE, but getOrdinate( 2 ) throws an exception.

Reply:
Thanks, fixed.